### ISSUE TITLE
Support for signatures without v value

### ISSUE THREAD

**Issue by jannikluhn** [ApeWorX/eth-keys#57]
### What was wrong?

The v4 identity scheme defined in the ENR spec uses signatures consisting of only `r` and `s`, not `v`. `eth-keys` currently does not allow verification of signatures without `v`.

### How can it be fixed?

Suggested approach:

- Implement [ecdsa_verify](https://github.com/ethereum/eth-keys/blob/2a97f2864a603ead280cd477ce94b54ce0199213/eth_keys/backends/base.py#L16-L20) in all backends without falling back to `ecdsa_recover`
- Allow instantiation of `Signature` with an `(r, s)` tuple and/or a 64 byte signature. All recover methods should raise some kind of exception (`ValidationError`?) if `v` is not set.

**Comment by jannikluhn** [comment]
Oh, and we probably need a method to create such a signature. Maybe `PrivateKey.sign_non_recoverable`?

**Comment by pipermerriam** [comment]
At the highest level I'm :-1: on this.

Wondering if we could segregate this into two classes so that rather than having APIs that *might* raise an error, you instead decide what type of signature you are working with and don't have to do any value checking.

`Signature` -> `RecoverableSignature`

We can figure out how to deal with names and the existing signature class if we like this approach.

**Comment by jannikluhn** [comment]
Agreed, two classes sound like the safer approach (probably with a shared base class).
